### PR TITLE
[GHA] Clarify some bits in the deps=high script

### PR DESCRIPTION
**Pull request by wouterj**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | Fix -
| License       | MIT
| Doc PR        | 

I had some difficulty understanding how flipped tests worked. Luckily, @nicolas-grekas was available to answer some questions. However, I'm sure that I've forgotten the logic in about a week time, so I think it's worth clarifying the code a bit (but feel free to close if it's just me).

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -184,24 +184,26 @@ jobs:
           # matrix.mode = high-deps
           echo "$COMPONENTS" | xargs -n1 | parallel -j +3 "_run_tests {} 'cd {} && $COMPOSER_UP && $PHPUNIT$LEGACY'" || X=1
 
+          # get a list of the patched components (relies on .github/build-packages.php being called in the previous step)
           (cd src/Symfony/Component/HttpFoundation; mv composer.bak composer.json)
-          COMPONENTS=$(git diff --name-only src/ | grep composer.json || true)
+          PATCHED_COMPONENTS=$(git diff --name-only src/ | grep composer.json || true)
 
-          if [[ $COMPONENTS && $SYMFONY_VERSION = *.4 ]]; then
+          # for the x.4 branch, checkout and test previous major with the patched components (only for patched components)
```
Comment (nicolas-grekas) [inline .github/workflows/unit-tests.yml]: for x.4 branches (there can be many, eg 4.4 & 5.4)

**Review by nicolas-grekas — APPROVED** [review]

File: .github/workflows/unit-tests.yml (hunk 1 above)
Comment (wouterj) [reply .github/workflows/unit-tests.yml]: fixed

**Review by chalasr — APPROVED** [review]
Thanks

**Review by OskarStark — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @wouterj.

**Pull request merged**